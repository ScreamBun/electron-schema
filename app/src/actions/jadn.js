import { ipcRenderer } from 'electron'

export const CONVERT_TO_JSON_SUCCESS = 'CONVERT_TO_JSON_SUCCESS';

export const convertToJSON = (schema) => {
  return (dispatch) => {
    ipcRenderer.invoke('render-python', {schema})
    .then(res => {
      dispatch(convertToJSONSuccess(res))
    });
  }

};

const convertToJSONSuccess = json_schema => ({
  type: CONVERT_TO_JSON_SUCCESS,
  payload: json_schema
});

