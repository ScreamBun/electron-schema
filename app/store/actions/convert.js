import { ipcRenderer } from 'electron';
import { SchemaFormats } from 'jadnschema';

// Store functions
export const CONVERT_TO_JSON_SUCCESS = '@@convert/CONVERT_TO_JSON_SUCCESS';
export const convertToJSON = schema => {
  return dispatch => ipcRenderer
    .invoke('convert-schema', {
      format: SchemaFormats.JSON,
      schema
    })
    // eslint-disable-next-line promise/always-return
    .then(jsonSchema => {
      dispatch({
        type: CONVERT_TO_JSON_SUCCESS,
        payload: JSON.parse(jsonSchema)
      });
    });
};


export const CONVERT_TO_MARKDOWN_SUCCESS = '@@convert/CONVERT_TO_MARKDOWN_SUCCESS';
export const convertToMD = schema => {
  return dispatch => ipcRenderer
    .invoke('convert-schema', {
      format: SchemaFormats.MarkDown,
      schema
    })
    // eslint-disable-next-line promise/always-return
    .then(mdSchema => {
      dispatch({
        type: CONVERT_TO_MARKDOWN_SUCCESS,
        payload: mdSchema
      });
    });
};
