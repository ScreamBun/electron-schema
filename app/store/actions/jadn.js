import { ipcRenderer } from 'electron';
import { SchemaFormats } from '../../src/utils';

// Helper functions
export const CONVERT_TO_JSON_SUCCESS = 'CONVERT_TO_JSON_SUCCESS';
const convertToJSONSuccess = jsonSchema => ({
  type: CONVERT_TO_JSON_SUCCESS,
  payload: jsonSchema
});

// Store functions
export const convertToJSON = schema => {
  return dispatch => ipcRenderer
    .invoke('convert-schema', {
      format: SchemaFormats.JSON,
      schema
    })
    // eslint-disable-next-line promise/always-return
    .then(res => {
      dispatch(convertToJSONSuccess(res));
    });
};
