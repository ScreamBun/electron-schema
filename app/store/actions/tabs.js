
// Store functions
export const SET_TABS_SUCCESS = '@@tabs/SET_TABS_SUCCESS';
export const setTabs = tabs => {
  return dispatch => {
    dispatch({
      type: SET_TABS_SUCCESS,
      payload: tabs
    });
  };
};


export const SET_ACTIVE_TAB_SUCCESS = '@@tabs/SET_ACTIVE_TAB_SUCCESS';
export const setActiveTab = tab => {
  return dispatch => {
    dispatch({
      type: SET_ACTIVE_TAB_SUCCESS,
      payload: tab.toLowerCase()
    });
  };
};
