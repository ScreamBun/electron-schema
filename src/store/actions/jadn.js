import { ipcRenderer } from 'electron';
import { SchemaFormats } from 'jadnschema';

const emit = (channel, ...args) => ipcRenderer.invoke(channel, ...args);

// Store functions
export const SET_BASE_JADN_SUCCESS = '@@jadn/SET_BASE_JADN_SUCCESS';
export const SET_BASE_JADN_FAILURE = '@@jadn/SET_BASE_JADN_FAILURE';
export const setJADN = schema => {
  return dispatch => emit('convert-schema', {
    format: SchemaFormats.JADN,
    schema
  })
  .then(rslt => {
    const payload = JSON.parse(rslt);
    const keys = Object.keys(payload);
    dispatch({
      // eslint-disable-next-line promise/always-return
      type: (keys.includes('info') && keys.includes('types')) ? SET_BASE_JADN_SUCCESS : SET_BASE_JADN_FAILURE,
      payload
    });
  });
};
