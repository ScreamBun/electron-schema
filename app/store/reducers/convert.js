import * as ConvertActions from '../actions/convert';

const initialState = {
  json_schema: {},
  md_schema: ''
};

export default (state = initialState, action = null) => {
  switch (action.type) {
    case ConvertActions.CONVERT_TO_JSON_SUCCESS:
      return {
        ...state,
        json_schema: action.payload
      };

    case ConvertActions.CONVERT_TO_MARKDOWN_SUCCESS:
      return {
        ...state,
        md_schema: action.payload
      };

    default:
      return state;
  }
};
