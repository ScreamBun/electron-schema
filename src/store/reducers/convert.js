import * as ConvertActions from '../actions/convert';

const initialState = {
  jadn_schema: {},
  jidl_schema: '',
  json_schema: {},
  md_schema: ''
};

export default (state = initialState, action = null) => {
  switch (action.type) {
    case ConvertActions.SET_BASE_JADN_SUCCESS:
      return {
        ...state,
        jadn_schema: action.payload
      };

    case ConvertActions.CONVERT_TO_JIDL_SUCCESS:
      return {
        ...state,
        jidl_schema: action.payload
      };

    case ConvertActions.CONVERT_TO_JSON_SUCCESS:
      return {
        ...state,
        json_schema: action.payload
      };

    case ConvertActions.CONVERT_TO_MARKDOWN_SUCCESS:
      return {
        ...state,
        md_schema: action.payload
      };

    default:
      return state;
  }
};
