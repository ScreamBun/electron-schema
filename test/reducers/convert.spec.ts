import { AnyAction } from 'redux';
import ConvertReducer from '../../app/store/reducers/convert';
import * as ConvertActions from '../../app/store/actions/convert';

describe('reducers', () => {
  describe('convert', () => {
    it('should handle initial state', () => {
      expect(ConvertReducer(undefined, {} as AnyAction)).toMatchSnapshot();
    });

    it('should handle SET_BASE_JADN_SUCCESS', () => {
      expect(
        ConvertReducer({ }, {
          type: ConvertActions.SET_BASE_JADN_SUCCESS,
          payload: {}
        })
      ).toMatchSnapshot();
    });

    it('should handle CONVERT_TO_JSON_SUCCESS', () => {
      expect(
        ConvertReducer({ }, {
          type: ConvertActions.CONVERT_TO_JSON_SUCCESS,
          payload: {}
        })
      ).toMatchSnapshot();
    });

    it('should handle CONVERT_TO_MARKDOWN_SUCCESS', () => {
      expect(
        ConvertReducer({ }, {
          type: ConvertActions.CONVERT_TO_MARKDOWN_SUCCESS,
          payload: ''
        })
      ).toMatchSnapshot();
    });

    it('should handle unknown action type', () => {
      expect(
        ConvertReducer({ }, { type: 'unknown' })
      ).toMatchSnapshot();
    });
  });
});
