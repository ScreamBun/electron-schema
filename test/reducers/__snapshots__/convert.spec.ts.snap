// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducers convert should handle CONVERT_TO_MARKDOWN_SUCCESS 1`] = `
Object {
  "md_schema": "",
}
`;

exports[`reducers convert should handle SET_BASE_JADN_SUCCESS 1`] = `
Object {
  "jadn_schema": Object {},
}
`;

exports[`reducers convert should handle SET_BASE_JADN_SUCCESS 2`] = `
Object {
  "jadn_schema": Object {},
}
`;

exports[`reducers convert should handle initial state 1`] = `
Object {
  "jadn_schema": Object {},
  "json_schema": Object {},
  "md_schema": "",
}
`;

exports[`reducers convert should handle unknown action type 1`] = `Object {}`;
